python/mlc_chat/nn/kv_cache.py

```diff
@@ -820,7 +820,11 @@ def _attention_decode(
     H_kv = num_kv_heads
     D = head_dim
 
-    thread_limit = 512 if str(target.kind) != "webgpu" else 256
+    max_threads = None
+    for name in ["max_threads_per_block", "max_num_threads"]:
+        if max_threads is None:
+            max_threads = target.attrs.get(name, None)
+    thread_limit = 512 if max_threads is None else max_threads
```

Comment:
Ooops sorry I approved early. We intended to use at most 512 threads here. So we will need to take `min(512, max_threads)` instead of overriding. Would you like to update it? I suspect this is the reason of CUDA CI failure.

Reply:
Thank you @MasterJH5574! Updated PR to check if it fixes the CI.